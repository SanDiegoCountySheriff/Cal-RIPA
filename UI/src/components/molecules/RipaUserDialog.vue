<template>
  <v-dialog
    v-model="modelDialog"
    max-width="650px"
    :light="getLight"
    :dark="getDark"
  >
    <v-card>
      <v-card-title>
        <span>Manage User</span>
      </v-card-title>

      <v-card-text>
        <ripa-officer v-model="modelStop"></ripa-officer>
      </v-card-text>

      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn color="blue darken-1" text @click="handleClose"> Cancel </v-btn>
        <v-btn color="blue darken-1" text @click="handleSave"> Save </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
import RipaOfficer from '@/components/molecules/RipaOfficer'

export default {
  name: 'ripa-user-dialog',

  components: {
    RipaOfficer,
  },

  data() {
    return {
      viewModelDialog: this.showDialog,
      viewModelStop: this.stop,
    }
  },

  computed: {
    modelDialog: {
      get() {
        return this.viewModelDialog
      },
      set(newValue) {
        if (!newValue) {
          if (this.onClose) {
            this.onClose()
          }
        }
        this.viewModelDialog = newValue
      },
    },

    modelStop: {
      get() {
        return this.viewModelStop
      },
      set(newValue) {
        this.viewModelStop = newValue
      },
    },

    getLight() {
      return this.$vuetify.theme.dark
    },

    getDark() {
      return !this.$vuetify.theme.dark
    },
  },

  methods: {
    handleClose() {
      if (this.onClose) {
        this.onClose()
      }
    },

    handleSave() {
      if (this.onSave) {
        this.onSave(this.viewModelStop)
      }
      this.handleClose()
    },
  },

  watch: {
    showDialog(newValue) {
      this.viewModelDialog = newValue
    },
    stop(newValue) {
      this.viewModelStop = newValue
    },
  },

  props: {
    showDialog: {
      type: Boolean,
      default: false,
    },
    stop: {
      type: Object,
      default: () => {},
    },
    onClose: {
      type: Function,
      default: () => {},
    },
    onSave: {
      type: Function,
      default: () => {},
    },
  },
}
</script>

<style lang="scss">
.v-dialog:not(.v-dialog--fullscreen) {
  max-height: 600px !important;
}
</style>
