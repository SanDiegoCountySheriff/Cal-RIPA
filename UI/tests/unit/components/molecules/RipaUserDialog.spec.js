import RipaUserDialog from '@/components/molecules/RipaUserDialog'
import { shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa User Dialog', () => {
  let wrapper
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  afterEach(() => {
    wrapper.destroy()
  })

  const factory = propsData => {
    return shallowMount(RipaUserDialog, {
      vuetify,
      propsData: {
        onClose: jest.fn(),
        onSave: jest.fn(),
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = factory({
      user: {
        agency: 'SDSD',
        assignment: 10,
        otherType: 'Data Services',
        startDate: '2014-10-10',
        yearsExperience: 7,
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
