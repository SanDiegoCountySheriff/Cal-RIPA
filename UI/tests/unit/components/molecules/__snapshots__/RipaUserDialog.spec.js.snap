// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa User Dialog should match snapshot 1`] = `
<div class="v-dialog__container" user="[object Object]" onclose="function () {
        return fn.apply(this, arguments);
      }" onsave="function () {
        return fn.apply(this, arguments);
      }">
  <!---->
</div>
`;
